handlers/validator.go

```diff
@@ -1729,32 +1765,60 @@ func ValidatorHistory(w http.ResponseWriter, r *http.Request) {
 			})
 			continue
 		}
-		events := template.HTML("")
-		if incomeDetails[index][i].AttestationSourcePenalty > 0 && incomeDetails[index][i].AttestationTargetPenalty > 0 {
-			events += utils.FormatAttestationStatusShort(2)
-		} else {
-			events += utils.FormatAttestationStatusShort(1)
+		event := types.ValidatorHistoryEvent{}
+
+		if incomeDetails[index][i].AttestationSourceReward > 0 {
+			event.AttestationSourceStatus = 1
+		} else if incomeDetails[index][i].AttestationSourcePenalty > 0 {
+			event.AttestationSourceStatus = 2
+		}
+
+		if incomeDetails[index][i].AttestationTargetReward > 0 {
+			event.AttestationTargetStatus = 1
+		} else if incomeDetails[index][i].AttestationTargetPenalty > 0 {
+			event.AttestationTargetStatus = 2
+		}
+
+		if incomeDetails[index][i].AttestationHeadReward > 0 {
+			event.AttestationHeadStatus = 1
 		}
 
 		if incomeDetails[index][i].ProposerAttestationInclusionReward > 0 {
-			block := utils.FormatBlockStatusShort(1)
-			events += block
+			event.BlockProposalStatus = 1
 		} else if incomeDetails[index][i].ProposalsMissed > 0 {
-			block := utils.FormatBlockStatusShort(2)
-			events += block
+			event.BlockProposalStatus = 2
+		}
+
+		if syncReward, syncPenalty := incomeDetails[index][i].SyncCommitteeReward, incomeDetails[index][i].SyncCommitteePenalty; syncReward > 0 {
+			if syncPenalty > 0 {
+				event.SyncParticipationStatus = 3
+				balanceChangeForSlot := (syncReward + syncPenalty) / utils.Config.Chain.Config.SlotsPerEpoch
+				event.SyncParticipationCount = utils.Config.Chain.Config.SlotsPerEpoch - (syncPenalty / balanceChangeForSlot)
+			} else {
+				event.SyncParticipationStatus = 1
+				event.SyncParticipationCount = utils.Config.Chain.Config.SlotsPerEpoch
+			}
+		} else if syncPenalty > 0 {
+			event.SyncParticipationStatus = 2
+		}
+
+		if incomeDetails[index][i].ProposerSlashingInclusionReward > 0 || incomeDetails[index][i].SlashingReward > 0 {
+			event.SlashingStatus = 1
+		}
+		if incomeDetails[index][i].SlashingPenalty > 0 {
+			event.SlashingStatus = 2
 		}
```

Comment:
Does this work?
I looked at a few validators that got slashed but none showed a `SlashingPenalty` and the icon remained grey.

Reply:
I already talked about this with @peterbitfly, it seems that CL rewards are calculated here https://github.com/gobitfly/eth-rewards/blob/main/ethrewards.go#L15 but the SlashingReward and SlashindPenalty fields are never set. I think for now it is okay to assume this will work down the line.